python/cudf_polars/cudf_polars/callback.py

```diff
@@ -61,6 +64,8 @@ def execute_with_cudf(
     try:
         with nvtx.annotate(message="ConvertIR", domain="cudf_polars"):
             nt.set_udf(partial(_callback, translate_ir(nt)))
-    except exception:
+    except exception as e:
+        if "POLARS_VERBOSE" in Config.state(if_set=True):
+            warnings.warn(f"cuDF execution failed: {type(e)}: {e}", stacklevel=2)
```

Comment:
Let's issue a polars `PerformanceWarning` here:
```
from polars.exceptions import PerformanceWarning
```

Let's also think about the message here. This is not failure of the execution, but rather this exception occurs if we couldn't translate the IR (which we claim is because we don't support it).

How about `f"Query execution with GPU not supported, reason: {type(e)}: {e}"`?

Reply:
Yes, I like this. Looks like this change has also already been made.